### PR TITLE
Task 7. (Player) New Class &  Task 8. (Player) availableSpace

### LINKED ISSUE #8: Task 7. (Player) New Class
Napisz klasę Player, która posiadać będzie 3 pola:

- std::shared_ptr<Ship> ship_ (dla chętnych, spróbuj użyć std::unique_ptr<>)
- money_
- availableSpace_

Dopisz także odpowiednie gettery oraz konstruktor.

Dopisz także 2 funkcje, które powinny zwracać dane ze statku:

- size_t getSpeed() const
- Cargo* getCargo(size_t index) const

### LINKED ISSUE #9: Task 8. (Player) availableSpace
W klasie Player napisz prywatną funkcję, która obliczać będzie availableSpace_ na podstawie aktualnej ilości towaru na statku.

### PR DESCRIPTION
**Pull request by Pawellos**
Nie wiem, czy nie namieszałem za bardzo, bo już się trochę pogubiłem z GITem i nie wiem czy dobrze zrozumiałem te 7 i 8  zadanie :)

### CODE REVIEW

File: shm/.gitignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Prerequisites
```
Comment (Kasperice) [inline shm/.gitignore:1]: You can delete this file from your branch. It will be merged in another PR

File: shm/cargo.cpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#include "cargo.hpp"
```
Comment (Kasperice) [inline shm/cargo.cpp:1]: You can delete this file from your branch. It will be merged in another PR

File: shm/cargo.hpp [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#pragma once
```
Comment (Kasperice) [inline shm/cargo.hpp:1]: You can delete this file from your branch. It will be merged in another PR

File: shm/ship.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#include "ship.hpp"
```
Comment (Kasperice) [inline shm/ship.cpp:1]: You can delete this file from your branch. It will be merged in another PR

File: shm/ship.hpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (Kasperice) [inline shm/ship.hpp:1]: You can delete this file from your branch. It will be merged in another PR

File: shm/player.hpp [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+
+size_t getShip() const;
+size_t getMoney() const; 
+size_t getAvailableSpace() const;
+
+size_t getSpeed(Ship ship) const;
+Cargo* getCargo(Ship ship, size_t index) const;
+
+};
```
Comment (Kasperice) [inline shm/player.hpp:28]: I would format it this way:
```suggestion
class Player{
   private:
    std::unique_ptr<Ship> ship_; 
    size_t availableSpace_;
    size_t money_;
    size_t CountAvailableSpace(Ship ship) const;
    
   public:
    Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace); 

    size_t getShip() const;
    size_t getMoney() const; 
    size_t getAvailableSpace() const;
    size_t getSpeed(Ship ship) const;
    
    Cargo* getCargo(Ship ship, size_t index) const;
};
```

File: shm/player.cpp [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#include "player.hpp"
+
+Player::Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace)
+ : ship_(std::move(ship)), money_(money), availableSpace_(availableSpace){}
+
+//size_t Player::getShip() const { return ship_; }
```
Comment (Kasperice) [inline shm/player.cpp:6]: Why this is commented?

File: shm/player.cpp [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+size_t Player::getAvailableSpace() const { return availableSpace_; }
+size_t Player::getMoney() const { return money_; }
+
+size_t Player::CountAvailableSpace(Ship ship) const {
+    size_t shipSpace = ship.getCapacity(); 
+    size_t wares = 0;
+
+    for(auto ware : ship.items_){
+    wares += ware.getAmount();
```
Comment (Kasperice) [inline shm/player.cpp:16]:
```suggestion
        wares += ware.getAmount();
```

File: shm/player.cpp [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    
+    return shipSpace - wares;
+}
+
+
+size_t Player::getSpeed(Ship ship) const{
+    return ship.getSpeed();
+}
+Cargo* Player::getCargo(Ship ship, size_t index) const{
```
Comment (Kasperice) [inline shm/player.cpp:26]:
```suggestion
}

Cargo* Player::getCargo(Ship ship, size_t index) const{
```

File: shm/player.cpp [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    size_t wares = 0;
+
+    for(auto ware : ship.items_){
+    wares += ware.getAmount();
+    }
+    
+    return shipSpace - wares;
+}
+
+
```
Comment (Kasperice) [inline shm/player.cpp:22]:
```suggestion

```

File: shm/player.cpp [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+size_t Player::getMoney() const { return money_; }
+
+size_t Player::CountAvailableSpace(Ship ship) const {
+    size_t shipSpace = ship.getCapacity(); 
+    size_t wares = 0;
+
+    for(auto ware : ship.items_){
+    wares += ware.getAmount();
+    }
+
```
Comment (Kasperice) [inline shm/player.cpp:18]:
```suggestion
```

File: shm/player.cpp [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    return shipSpace - wares;
+}
+
+
+size_t Player::getSpeed(Ship ship) const{
+    return ship.getSpeed();
+}
+Cargo* Player::getCargo(Ship ship, size_t index) const{
+    Cargo * ware = &ship.items_[index];
+    return ware;
```
Comment (Kasperice) [inline shm/player.cpp:28]: How does it work?

**Review by Kasperice — CHANGES_REQUESTED** [review]

File: shm/player.hpp [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+size_t CountAvailableSpace(Ship ship) const;
+public:
+
+Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace); 
+
+
+size_t getShip() const;
+size_t getMoney() const; 
+size_t getAvailableSpace() const;
```
Comment (Kasperice) [inline shm/player.hpp:23]: as @g00fer said on Discord, lets use something else than size_t for values/functions like those (this comment applies to all player files)

**Review by Kasperice — CHANGES_REQUESTED** [review]

**Comment by Kasperice** [comment]
Can we close this PR? I saw that you created second one

File: shm/cargo.cpp [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+
+Cargo& Cargo::operator+=(size_t amount) {
```
Comment (g00fer) [inline shm/cargo.cpp:3]: do not use size_t. See my comments on discord suad-7 channel on that

**Review by g00fer — CHANGES_REQUESTED** [review]
Do not use size_t (see my comment on squad-7 channel). Make sure it is compiling.

**Pull request closed without merging**